library/src/tensile_host.cpp

```diff
@@ -405,30 +409,49 @@ TensileHost* createTensileHost()
  * runContractionProblem calls Tensile to run a contraction problem described *
  * by RocblasContractionProblem                                               *
  ******************************************************************************/
-//
 template <typename Ti, typename To, typename Tc>
 rocblas_status
     TensileHost::runContractionProblem(const RocblasContractionProblem<Ti, To, Tc>& problem)
-try
 {
-    // We know that the TensileHost instance is a TensileHostImpl, so we can downcast to it
-    auto* host            = static_cast<TensileHostImpl*>(this);
-    auto  tensile_problem = ConstructTensileProblem(problem);
-    auto  inputs          = GetTensileInputs(problem);
-    auto  solution        = host->library->findBestSolution(tensile_problem, *host->hardware);
-    if(!solution)
+    std::shared_ptr<Tensile::ContractionSolution> solution;
+    rocblas_status                                status = rocblas_status_internal_error;
+
+    try
+    {
+        // We know that the TensileHost instance is a TensileHostImpl, so we can downcast to it
+        auto* host            = static_cast<TensileHostImpl*>(this);
```

Comment:
The base class is TensileHost so this upcast should use dynamic_cast which would correctly fail if you change the classes.

Reply:
We do not need to use dynamic_cast.

These are not polymorphic classes.

These are PIMPL classes, which can use static_cast for better performance and smaller code.

Note that you are reviewing code that's already checked into develop; this PR just refined some of it.

This will not be changed in future releases. This design is intentional.